TradeItIosTicketSDK2/TradeService.swift

```diff
@@ -1,26 +1,47 @@
-internal class TradeItTradeService: NSObject {
-    private let session: TradeItSession
+protocol TradeService {
+    associatedtype PreviewTradeRequest: TradeItAuthenticatedRequest
+    associatedtype PreviewTradeResult: TradeItResult
+    associatedtype PlaceTradeResult: TradeItResult
+    static var previewTradeEndpoint: String { get }
+    static var placeTradeEndpoint: String { get }
+    var session: TradeItSession { get }
 
-    init(session: TradeItSession) {
-        self.session = session
-    }
+    init(session: TradeItSession)
 
     func previewTrade(
-        _ data: TradeItPreviewTradeRequest,
-        onSuccess: @escaping (TradeItPreviewTradeResult) -> Void,
+        _ data: Self.PreviewTradeRequest,
+        onSuccess: @escaping (Self.PreviewTradeResult) -> Void,
+        onFailure: @escaping (TradeItErrorResult) -> Void
+    )
+
+    func placeTrade(
+        _ data: TradeItPlaceTradeRequest,
+        withCompletionBlock completionBlock: @escaping (TradeItResult) -> Void
+    )
+
+    func answerSecurityQuestionPlaceOrder(
```

Comment:
`...PlaceOrder` vs `...placeTrade` is there a reason for this difference in vernacular?

Reply:
It diverged in the past. Can update but would rather do that as a separate chore as this PR is too big.